cmake/rigel.cmake

```
@@ -0,0 +1,28 @@
+# Emscripten provides its own versions of the SDL and Boost libraries.
+# Calling this function defines the same targets that would normally be
+# defined by the corresponding find_package calls, but these targets
+# set the appropriate compiler and linker flags for Emscripten instead of
+# referring to external libraries.
+function(rigel_define_wasm_targets_for_dependencies)
+    add_library(SDL2::Core INTERFACE IMPORTED)
+    set_target_properties(SDL2::Core PROPERTIES
+        INTERFACE_COMPILE_OPTIONS
+        "SHELL:-s USE_SDL=2"
+        INTERFACE_LINK_OPTIONS
+        "SHELL:-s USE_SDL=2"
+    )
```

```suggestion
    target_compile_options(SDL2::Core INTERFACE "SHELL:-s USE_SDL=2")
    target_link_options(SDL2::Core INTERFACE "SHELL:-s USE_SDL=2")
```
and similarly for `SDL2::Mixer` and `Boost::boost`.

---

Yeah that looks much more readable, thanks!